src/Sarif.Viewer.VisualStudio.Core/OutputWindowTracerListener.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.Sarif.Viewer
+{
+    public class OutputWindowTracerListener : TraceListener
+    {
+        private readonly string _name;
+        private readonly IVsOutputWindow _outputWindowService;
+
+        private IVsOutputWindowPane pane;
+
+        public OutputWindowTracerListener(IVsOutputWindow outputWindowService, string name)
+        {
+            this._outputWindowService = outputWindowService;
+            this._name = name;
+            Trace.Listeners.Add(this);
+        }
+
+        public override void Write(string message)
+        {
+            if (!SarifViewerPackage.IsUnitTesting)
+            {
+#pragma warning disable VSTHRD108 // Assert thread affinity unconditionally
+                ThreadHelper.ThrowIfNotOnUIThread();
+#pragma warning restore VSTHRD108
+            }
```

Comment:
This whole block of code is repeated three times and isn't clear what it is doing. How about carving this out to a new method (called three times), name it something helpful that describes the purpose/function. And then add any additional needed comments strictly in the one method. This should clean up the code and help explain exactly what this is doing.

Reply:
Unfortunately the `ThreadHelper.ThrowIfNotOnUIThread()` has to be called in the same function the UI components are used. Moving it to a common function does not eliminate the build error.